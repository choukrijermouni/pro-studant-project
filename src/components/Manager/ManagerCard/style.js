import { blue } from '@pro_boa/ui'
export default {
  row: {
    marginBottom: 16,
    display: 'flex',
    justifyContent: 'flex-start',
    minHeight: 50,
    overflow: 'hidden',
    alignItems: 'center',
    padding: 12
  },
  checkBox: {
    width: '0px !important'
  },
  card: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    width: '11vw'
  },
  icon: {
    fontSize: 16,
    color: blue[1]
  },
  teamTag: {
    width: 43,
    height: 43,
    backgroundColor: blue[0],
    borderRadius: '50%',
    marginLeft: 16
  },
  managerCard: {
    display: 'flex',
    alignItems: 'center',
    marginLeft: 24
  },
  flex: {
    display: 'flex',
    alignItems: 'center'
  },
  avatarContainer: {
    transition: 'transform 0.2s ease'
  },
  noTeamMessageClass: {
    display: 'flex',
    alignItems: 'center'
  },
  nameContainer: {
    width: '15vw',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    marginRight: 25
  },
  iconSkeleton: {
    display: 'flex',
    width: 30
  },
  spaceSekelton: {
    padding: '20px 2px',
    height: 68
  },
  avatarspaceSkeleton: {
    padding: '20px 2px',
    height: 68
  },
  deleteStyle: {
    position: 'absolute',
    right: '2rem',
    top: '2rem'
  },
  inviteStyle: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center'
  }
}
