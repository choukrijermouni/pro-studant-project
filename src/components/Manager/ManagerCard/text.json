{
  "noTeamMessage": "N’appartient à aucune équipe",
  "neverConnected": "Aucune activité",
  "userInvited": "Invité par e-mail à l'adresse : ",
  "creation": " Date de création",
  "lastConnection": "Date de dernière connexion",
  "team": "Équipes",
  "cancelInvitation": "Annuler l'invitation"
}