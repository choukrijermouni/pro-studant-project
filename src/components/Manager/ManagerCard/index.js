import React, { useState } from 'react'
import { createUseStyles } from 'react-jss'
import style from './style'
import { Paper, TypographyElement, neutral, Icon, Icons, Avatar, AvatarSize, red, Skeleton } from '@pro_boa/ui'
import moment from 'moment'
import Text from './text.json'
import { avatarName } from 'helpers'
import { amazonBucket, DMYDateFormat, Avatars } from 'constants/'
import DeleteInvitationModal from 'components/Common/InviteBar/DeleteInvitationModal'
import { useDispatch, useSelector } from 'react-redux'
import { cancelLearnerInvitationAction } from 'pages/LearnerProfile/store'

const numberOfTeams = 3
const defaultHandleClick = () => { }
export default ({ handleClick = defaultHandleClick, Data = [], loading }) => {
  const useStyle = createUseStyles(style)
  const {
    row,
    card,
    icon,
    flex,
    avatarContainer,
    managerCard,
    noTeamMessageClass,
    nameContainer,
    avatarspaceSkeleton,
    spaceSekelton,
    iconSkeleton,
    deleteStyle,
    inviteStyle
  } = useStyle()
  const dispatch = useDispatch()
  const [openDeleteInvitationModal, setOpenDeleteInvitationModal] = useState(false)
  const { InvitationId } = useSelector(({ managers }) => managers?.invitedManager)

  const fullName = `${Data.FirstName} ${Data.LastName}`
  const teams = Data?.Teams
  const isInvited = Data?.InvitationId
  return (
    loading
      ? (
        <Paper className={row}>
          <div className={flex} onClick={handleClick}>
            <div className={nameContainer}>
              <div className={avatarContainer}>
                <Avatar
                  loading
                  size={AvatarSize.size_1}
                />
              </div>
              <span className={avatarspaceSkeleton} />
              <Skeleton lines={1} height={18} width={150} />
            </div>
            <div title={Text.creation} className={card}>
              <div className={iconSkeleton}>
                <Skeleton lines={1} height={18} width={18} />
                &nbsp;
              </div>
              <span className={spaceSekelton} />
              <Skeleton lines={1} height={18} width={75} />
            </div>
            <div title={Text.lastConnection} className={card}>
              <div className={iconSkeleton}>
                <Skeleton lines={1} height={18} width={18} />
                &nbsp;
              </div>
              <span className={spaceSekelton} />
              <Skeleton lines={1} height={18} width={100} />
            </div>
            <div title={Text.team} className={card}>
              <div className={iconSkeleton}>
                <Skeleton lines={1} height={18} width={18} />
                &nbsp;
              </div>
              <span className={spaceSekelton} />
              <Skeleton lines={1} height={18} width={81} />
            </div>
          </div>
        </Paper>
        )
      : (
        <>
          {
            isInvited
              ? (
                <Paper className={row}>
                  <div className={flex} onClick={handleClick}>
                    <div className={inviteStyle}>
                      <div className={flex}>
                        <div className={avatarContainer}>
                          <Avatar
                            dotTop={1}
                            name={avatarName(fullName)}
                            img={`${amazonBucket.avatar}${Avatars.anonymous}`}
                            size={AvatarSize.size_1}
                            dot
                            dotColor={red[3]}
                          />
                        </div>
                        <TypographyElement
                          component='h3'
                          variant='heading3'
                          align='left'
                          fontSize={16}
                          spacing='20px'
                          color={neutral[6]}
                        >
                          {Text.userInvited} {Data.InvitationEmail}
                        </TypographyElement>
                      </div>
                      <div>
                        <TypographyElement
                          color={red[3]}
                          spacing='0px 20px 0 20px'
                          cursor='pointer'
                          fontSize={12}
                          className={deleteStyle}
                          handleClick={() => setOpenDeleteInvitationModal(true)}
                        >
                          {Text.cancelInvitation}
                        </TypographyElement>
                      </div>
                    </div>
                  </div>
                  <DeleteInvitationModal
                    isManager
                    openModal={openDeleteInvitationModal}
                    handleClose={() => setOpenDeleteInvitationModal(false)}
                    handleSubmit={() => {
                      dispatch(cancelLearnerInvitationAction(InvitationId))
                      setOpenDeleteInvitationModal(false)
                    }}
                  />
                </Paper>
                )
              : (
                <Paper className={row}>
                  <div className={flex} onClick={handleClick}>
                    <div className={nameContainer}>
                      <div className={avatarContainer}>
                        <Avatar
                          name={avatarName(fullName)}
                          img={Data.Photo ? `${amazonBucket.bucketBaseUrl}${Data.Photo}` : null}
                          size={AvatarSize.size_1}
                        />
                      </div>
                      <TypographyElement
                        component='h3'
                        variant='heading3'
                        align='left'
                        fontSize={16}
                        spacing='20px'
                        color={neutral[6]}
                      >
                        {Data.FirstName || ''} {Data.LastName || ''}
                      </TypographyElement>
                    </div>
                    <div title={Text.creation} className={card}>
                      <Icon iconName={Icons.calendar} style={icon} />
                      <TypographyElement
                        component='h4'
                        variant='caption1'
                        align='left'
                        spacing='10px 16px 11px 16px'
                        color={neutral[4]}
                      >
                        {moment(Data.CreationDate).format(DMYDateFormat)}
                      </TypographyElement>
                    </div>
                    <div title={Text.lastConnection} className={card}>
                      <Icon iconName={Icons.lastUserLogin} style={icon} />
                      <TypographyElement
                        component='h4'
                        variant='caption1'
                        align='left'
                        spacing='10px 16px 11px 16px'
                        color={neutral[4]}
                      >
                        {Data.LastConnectionDate !== null
                          ? `${moment(Data.LastConnectionDate).format(DMYDateFormat)}`
                          : Text.neverConnected}
                      </TypographyElement>
                    </div>
                    <div title={Text.team} className={managerCard}>
                      {teams?.length
                        ? (
                          <>
                            <Icon iconName={Icons.teams} style={icon} />
                            <TypographyElement
                              component='h4'
                              variant='caption1'
                              align='left'
                              spacing='10px 5px 11px 16px'
                              color={neutral[4]}
                            >
                              {teams.map(team => team.Name.length > 15 ? `${team.Name.slice(0, 15)}...` : team.Name)
                                .slice(0, numberOfTeams).join(' - ')}
                              {teams.length > numberOfTeams && ' . . .'}
                            </TypographyElement>
                          </>
                          )
                        : (
                          <div title={Text.team} className={noTeamMessageClass}>
                            <Icon iconName={Icons.teams} style={icon} />
                            <TypographyElement
                              component='h4'
                              variant='caption1'
                              align='left'
                              spacing='10px 16px 11px 16px'
                              color={neutral[4]}
                            >
                              {Text.noTeamMessage}
                            </TypographyElement>
                          </div>
                          )}
                    </div>
                  </div>
                </Paper>
                )
          }
        </>
        ))
}
