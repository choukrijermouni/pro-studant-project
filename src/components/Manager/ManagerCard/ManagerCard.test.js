import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import ManagerCard from '.'
configure({ adapter: new Adapter() })

describe('>>> ManagerCard --- Shallow Render ManagerCard component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<ManagerCard />)
    tree = renderer.create(
      <ManagerCard />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
