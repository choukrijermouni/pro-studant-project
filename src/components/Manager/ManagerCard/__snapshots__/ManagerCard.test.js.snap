// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> ManagerCard --- Shallow Render ManagerCard component +++ render correctly and match snapshot 1`] = `
<div>
  <div
    className="row-0-2-6 paper-0-2-11"
  >
    <div
      className="inputGroup-0-2-12"
    >
      <input
        className="input-0-2-14"
        disabled={false}
        id="desktop1"
        type="checkbox"
      />
      <label
        className="checkbox-0-2-13 checkbox-d0-0-2-16"
      />
      <label
        className="label-0-2-15 checkBox-0-2-7"
        htmlFor="desktop1"
      >
        
      </label>
    </div>
    <div
      className="avatarContainer-0-2-18"
    >
      <div
        className="dot-0-2-17"
      />
      <span
        className="avatarGroup-0-2-19 avatarGroup-d0-0-2-20"
      >
        YS
      </span>
    </div>
    <p
      className="text-0-2-21 text-d0-0-2-22"
    >
      choukri jermouni
    </p>
    <div
      className="card-0-2-8"
    >
      <span
        className="icon-creation-date-alph-iconfont icon-0-2-9"
      />
      <p
        className="text-0-2-23 text-d0-0-2-24"
      >
        01/01/2020
      </p>
    </div>
    <div
      className="card-0-2-8"
    >
      <span
        className="icon-last-login-user-alph-iconfont icon-0-2-9"
      />
      <p
        className="text-0-2-25 text-d0-0-2-26"
      >
        01/01/2020
      </p>
    </div>
    <div
      className="card-0-2-8"
    >
      <span
        className="icon-temas-alph-iconfont icon-0-2-9"
      />
      <p
        className="text-0-2-27 text-d0-0-2-28"
      >
        DÃ©veloppement, Design
      </p>
    </div>
  </div>
</div>
`;
